src/main/java/com/org/jenkins/custom/jenkins/distribution/service/PackagerController.java

```diff
@@ -34,4 +43,22 @@
             return new ResponseEntity(HttpStatus.NOT_FOUND);
         }
     }
+
+    /**
+     * @param postPayload Cofiguration on the front end that gets posted as a string
+     * @return Packager-config.yml
+     */
+    @PostMapping (path = "/downloadPackageConfiguration")
+    public ResponseEntity<?> downloadPackageConfig(@RequestBody String postPayload) {
```

Comment:
What's the difference between "downloading the package configuration" and "getting the package configuration"? I am referring to the other method above. They both return the same type, but I am guessing that the content is different. Can you please clarify in the javadoc?

Reply:
Oh yes one just generates the package configuration hence the `getPackageConfiguration` so the server is like get me the package configuration and the other method is downloading the package configuration hence the name `downloadPackageConfiguration`

Reply:
Clarified in javadoc

Reply:
@sladyn98 I think my confusion comes from the name of the methods. Both methods appear to return to the user the same thing: a "PackageConfig". Do both methods return they the same thing to the user?

Reply:
Yes the download method lets you download the pacakge config file whereas the getPackage returns only the package in the form of a string

Reply:
After our discussion, I understand the difference. Thanks!

Reply:
I think this is a good thing to add into the comments to help explain what's going on with these 2 methods.

Reply:
Perfect Added :+1: